data/org.eclipse.birt.report.data.oda.jdbc/src/org/eclipse/birt/report/data/oda/jdbc/JDBCDriverManager.java

```diff
@@ -313,6 +317,31 @@ private Connection doConnect(String driverClass, String url, String jndiNameUrl,
 		}
 	}
 
+	/*
+	 * Remove unsupported properties from the connection properties
+	 * based on the driver property information set
+	 */
+	private void removeUnsupportedConnectionProperties(Driver driver, String url, Properties connectionProperties) {
+		try {
+			DriverPropertyInfo[] supportedProperties = driver.getPropertyInfo(url, connectionProperties);
+			if (supportedProperties != null && connectionProperties != null) {
+				connectionProperties.forEach((key, value) -> {
+					boolean isPropertySupported = false;
+					for (DriverPropertyInfo supportedProperty : supportedProperties) {
+						if (supportedProperty.name.equalsIgnoreCase(key.toString())) {
+							isPropertySupported = true;
+							break;
+						}
+					}
+					// remove unsupported connection properties
+					if (!isPropertySupported)
+						connectionProperties.remove(key);
```

Comment:
Isn't this inside the connectionProperties' forEach such that is will cause a concurrent modification exception?

Reply:
Of course, it is not a good style to remove elements of an List during the looping himself.
I can also create a name-key list and remove the unsupported keys after the serach loop separately.

Reply:
Changed to use registry-list and removement of properties on an own remove-loop